### PR TITLE
[fix](types)throw exception if complex type in text table

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

For Hive text input format:  the column types ARRAY/MAP/STRUCT are not supported yet.
It will be supported over successive versions.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java [hunk 1]
```diff
@@ -293,6 +296,17 @@ private void initHMSExternalTable(HMSExternalTable hmsTable) throws UserExceptio
                 scanProvider = new IcebergScanProvider(hmsSource, analyzer);
                 break;
             case HIVE:
+                String inputFormat = hmsTable.getRemoteTable().getSd().getInputFormat();
+                if (inputFormat.contains("TextInputFormat")) {
+                    for (SlotDescriptor slot : desc.getSlots()) {
+                        if (slot.getType() instanceof ArrayType
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java:302]: use `slot.getType().isScalarType()`

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java [hunk 2]
```diff
@@ -293,6 +296,17 @@ private void initHMSExternalTable(HMSExternalTable hmsTable) throws UserExceptio
...
+                                || slot.getType() instanceof MapType
+                                || slot.getType() instanceof StructType) {
+                            throw new UserException("The column types ARRAY/MAP/STRUCT are not supported yet"
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java:305]: show the column name in error log

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.2 seconds
 stream load tsv:          460 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230322095523_clickbench_pr_118577.html

**Comment by wsjz** [comment]
run buildall

**Pull request merged**